lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetMetadataUtils.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.writer;
+
+import org.apache.parquet.column.statistics.BinaryStatistics;
+import org.apache.parquet.format.Statistics;
+import org.apache.parquet.internal.column.columnindex.BinaryTruncator;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.ColumnOrder;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+// Based on org.apache.parquet.format.converter.ParquetMetadataConverter
+public class ParquetMetadataUtils
+{
+    private static final long MAX_STATS_SIZE = 4096; // limit stats to 4k
```

Comment:
Shouldn't we keep the limit up to `4096` independently of what the `truncateLength` value is?

Reply:
Now that we're truncating min and max to 1024 length, this is overall 4K limit is not needed

Reply:
I'm thinking more for to guard for eventual future use-cases where the `truncateLength` could be potentially much higher.

Reply:
It's confusing to have another limit on top of truncation length. Since there's only 2 values (min and max), using a sensible value for truncation length is good enough to keep overall stats size in check.